networkx/classes/backends.py

```diff
@@ -107,44 +107,78 @@ def _register_algo(name, wrapped_func):
     wrapped_func.dispatchname = name
 
 
-def _dispatch(func=None, *, name=None):
+def _dispatch(func=None, *, name=None, graphs=None):
     """Dispatches to a backend algorithm
     when the first argument is a backend graph-like object.
+
+    The algorithm name is assumed to be the name of the wrapped function unless
+    `name` is provided. This is useful to avoid name conflicts, as all
+    dispatched algorithms live in a single namespace.
+
+    If more than one graph is required for the algorithm, provide a comma-separated
+    string of variable names as `graphs`. These must be the same order and name
+    as the variables passed to the algorithm. Dispatching does not support graphs
+    which are not the first argument(s) to an algorithm.
     """
     # Allow any of the following decorator forms:
     #  - @_dispatch
     #  - @_dispatch()
-    #  - @_dispatch("override_name")
     #  - @_dispatch(name="override_name")
+    #  - @_dispatch(graphs="G,H")
+    #  - @_dispatch(name="override_name", graphs="G,H")
     if func is None:
-        if name is None:
+        if name is None and graphs is None:
             return _dispatch
-        return functools.partial(_dispatch, name=name)
+        return functools.partial(_dispatch, name=name, graphs=graphs)
     if isinstance(func, str):
-        return functools.partial(_dispatch, name=func)
+        raise TypeError("'name' and 'graphs' must be passed by keyword") from None
     # If name not provided, use the name of the function
     if name is None:
         name = func.__name__
+    # If graphs not provided, use the most common signature of a single graph named `G`
+    if graphs is None:
+        graphs = "G"
```

Comment:
Would the default value be more clear as `graphs='G'`?  The use of `None` as an indicator of the default value is helpful for mutable objects (so all function calls don't use the same default object). But here the decorator signature to be more clear with the default value stated explicitly.  And we can eliminate any use with `_dispatch(graphs="G")`.  I don't feel strongly -- just a suggestion.   (and maybe I am missing something because above you simply return `_dispatch` which presumably has `graphs=None`...

Reply:
Good suggestion. I've updated the code to use the simpler string default.